library/bignum_mod_raw.h

```diff
@@ -233,6 +232,24 @@ int mbedtls_mpi_mod_raw_to_mont_rep( mbedtls_mpi_uint *X,
  */
 int mbedtls_mpi_mod_raw_from_mont_rep( mbedtls_mpi_uint *X,
                                        const mbedtls_mpi_mod_modulus *m );
+
+/** \brief  Perform fixed width modular negation.
+ *
+ * The size of the operation is determined by \p N. \p A must have
+ * the same number of limbs as \p N.
+ *
+ * \p X may be aliased to \p A.
+ *
+ * \param[out] X        The result of the modular negation.
+ *                      This must be initialized. Must have enough limbs to
+ *                      store the full value of the result.
```

Comment:
I don't like the phrasing "must have enough limbs". mod_raw should be working with fixed-size MPIs.

For example: if X has more limbs than A, only the number of limbs of A are set, so if X is then treated as a number with more limbs than A the result will be wrong.

Therefore I think we should be saying "A and X must have the same number of limbs as the modulus".

Hopefully the work that @wernerlewis is doing on the mod_raw conventions will help here.